### PR TITLE
Refactor dialog topic and add api summary

### LINKED ISSUE #124: Create Dialog Windows user story and update current topic based on the article guidance.
Article guidance - https://github.com/IgniteUI/igniteui-docfx/blob/master/.github/CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by sstoyanovIG**
Closes #124

### CODE REVIEW

**Comment by kdinev** [comment]
@ig-ealbert Stefan has overwritten some of the JP content, where it gets updated. Can you double check though?

File: en/components/dialog.md [hunk 1]
```diff
@@ -4,87 +4,180 @@ _description: Whether building informative dialogs or data manipulation windows,
...
+</div>
+<div class="divider--half"></div>
 
-`leftButtonBackgroundColor`
+####Custom Dialog
 
-`leftButtonRipple`
-<div class="divider--half"></div>
+To add custom dialog, in the template of our sign in component we can add the following code to get the custom dialog. We have to set the `title`,`leftButtonLabel`,`rightButtonBackgroundColor`, `rightButtonColor`, `rightButtonLabel`, `closeOnOutsideSelect` and handle `onLeftButtonSelect` and `onRightButtonSelect` event.
+Also we can add two groups of label and input decorated with the [**igxLabel**](../components/label_input.html) and [**igxInput**](../components/label_input.html) directives.
```
Comment (ig-ealbert) [inline en/components/dialog.md]: I've now seen links done in 3 ways in the topics:
1. Production link.
2. Relative link that is just the name of the topic.
3. Relative link as is used here.
Which is correct?  Do all of them work?  Should it be consistent?

File: en/components/dialog.md [hunk 2]
```diff
@@ -4,87 +4,180 @@ _description: Whether building informative dialogs or data manipulation windows,
...
+#### Inputs
+The following inputs are available in the **igx-dialog** component:
+
+| Name | Type | Description |
+| :--- | :--- | :--- |
+| `id` | string | Unique identifier of the dialog. |
+| `title` | string | Specifies the title of the dialog. |
+| `message` | string | Specifies the message of the dialog. |
+| `leftButtonLabel` | string | When set, the left button becomes visible. |
+| `leftButtonType` | string | Specifies the left button [**type**](button.md#button-types). Default value is flat. |
```
Comment (ig-ealbert) [inline en/components/dialog.md]: I'm pretty sure you can't link to .md files in production.

File: jp/components/dialog.md [hunk 3]
```diff
@@ -1,99 +1,183 @@
 ﻿---
-title: Dialog Window コンポーネント
-_description: Ignite UI for Angular Dialog Window コンポーネントを使用すると、情報ダイアログまたはデータ変換ウィンドウを作成し、リアルタイムで情報を表示して管理できます。
-_keywords: Ignite UI for Angular, UI コントロール, Angular ウィジェット, web ウィジェット, UI ウィジェット, Angular, ネイティブ Angular コンポーネント スィート, ネイティブ Angular コントロール, ネイティブ Angular コンポーネント ライブラリ, Angular Dialog Window コンポーネント, Angular Dialog Window コントロール
-_language: ja
```
Comment (ig-ealbert) [inline jp/components/dialog.md]: Did we stop using this metadata field?

File: en/components/dialog.md (hunk 1 above)
Comment (sstoyanovIG) [reply en/components/dialog.md]: The correct way is to use .md files. I will update these ones.

File: en/components/dialog.md (hunk 2 above)
Comment (sstoyanovIG) [reply en/components/dialog.md]: Basically there is a parser, which converts the .md files to html files, so this should work on production

File: jp/components/dialog.md (hunk 3 above)
Comment (sstoyanovIG) [reply jp/components/dialog.md]: I noticed none of the topics has this field, may be @simeonoff know more?

Comment (simeonoff) [reply jp/components/dialog.md]: These are required fields, each article should include them. The values for _description and _keywords are provided by marketing. Simply put - you shouldn't be deleting them.

File: en/components/dialog.md (hunk 2 above)
Comment (zdrawku) [reply en/components/dialog.md]: We are going to review the changes after merging them to staging.

File: jp/components/dialog.md (hunk 3 above)
Comment (ig-ealbert) [reply jp/components/dialog.md]: @simeonoff I meant particularly the `_language` metadata field.  That is the one that is being removed in some of the new topics.  I think it's a copy/paste error as I've noticed the developer copying the English topic into the Japanese file.  I have been fixing this as I add the Japanese translations but it's error prone as I have to remember to do it.

Comment (ig-ealbert) [reply jp/components/dialog.md]: Not sure if it needs to be in there for the English topic.  Mostly interested if it is still used in the Japanese topics to know if I should continue adding it in there.

**Pull request merged**